static/app/views/projectEventRedirect.tsx

```diff
@@ -1,73 +1,125 @@
+import useOrganization from 'sentry/utils/useOrganization';
 import {useParams} from 'sentry/utils/useParams';
+import {getTraceDetailsUrl} from 'sentry/views/performance/traceDetails/utils';
 
 /**
- * This component performs a client-side redirect to Event Details given only
- * an event ID (which normally additionally requires the event's Issue/Group ID).
- * It does this by using an XHR against the identically-named ProjectEventRedirect
- * _Django_ view, which responds with a 302 with the Location of the corresponding
- * Event Details page (if it exists).
+ * This component redirects to the Event Details page given only an event ID
+ * (which normally additionally requires the event's Issue/Group ID).
  *
- * See:
- * https://github.com/getsentry/sentry/blob/824c03089907ad22a9282303a5eaca33989ce481/src/sentry/web/urls.py#L578
+ * It fetches the event data from the API to extract the group ID, then navigates
+ * to the appropriate issue event page. For events without a group ID (e.g.,
+ * transaction events), it falls back to the trace details page.
+ *
+ * This component handles routes:
+ * - /projects/:projectId/events/:eventId/
+ * - /:orgId/:projectId/events/:eventId/ (legacy)
  */
 export default function ProjectEventRedirect() {
-  const [error, setError] = useState<string | null>(null);
+  const organization = useOrganization();
+  const location = useLocation();
   const navigate = useNavigate();
-  const params = useParams<{eventId: string; orgId: string; projectId: string}>();
+  const params = useParams<{eventId: string; projectId: string}>();
+  const datetimeSelection = normalizeDateTimeParams(location.query);
+
+  // Construct the eventSlug in the format expected by the events API: "projectSlug:eventId"
+  const eventSlug = `${params.projectId}:${params.eventId}`;
+
+  const {
+    data: event,
+    isPending,
+    error,
+  } = useApiQuery<Event>(
+    [`/organizations/${organization.slug}/events/${eventSlug}/`],
+    {staleTime: 2 * 60 * 1000} // 2 minutes in milliseconds
+  );
 
   useEffect(() => {
-    // This presumes that _this_ React view/route is only reachable at
-    // /:org/:project/events/:eventId (the same URL which serves the ProjectEventRedirect
-    // Django view).
-    const endpoint = `/organizations/${params.orgId}/projects/${params.projectId}/events/${params.eventId}/`;
+    if (!event) {
+      return;
+    }
 
-    // Use XmlHttpRequest directly instead of our client API helper (fetch),
-    // because you can't reach the underlying XHR via $.ajax, and we need
-    // access to `xhr.responseURL`.
-    //
-    // TODO(epurkhiser): We can likely replace tihs with fetch
-    const xhr = new XMLHttpRequest();
+    // If the event has a group ID, navigate to the issue event page
+    if (event.groupID && event.eventID) {
+      navigate(
+        {
+          pathname: `/organizations/${organization.slug}/issues/${event.groupID}/events/${event.eventID}/`,
+          query: {
```

Comment:
Preserves only the `project` & `referrer` query params, was previously just `project`